ImageStreamIO.c

```diff
@@ -343,19 +328,17 @@ errno_t ImageStreamIO_filename(
 
     int rv = snprintf(file_name, ssz, "%s/%s.im.shm", shmdirname, im_name);
 
-    if ((rv > 0) && (rv < (int)ssz))
-    {
-        return IMAGESTREAMIO_SUCCESS;
-    }
-    else if (rv < 0)
+    if ((rv > 0) && (rv < (int)ssz)) { return IMAGESTREAMIO_SUCCESS; }
+
+    if (rv < 0)
     {
         ImageStreamIO_printERROR(IMAGESTREAMIO_FAILURE, strerror(errno));
         return IMAGESTREAMIO_FAILURE;
     }
     else
     {
         ImageStreamIO_printERROR(IMAGESTREAMIO_FAILURE,
-                                 "string not large enough for file name");
+                                 (char*)"string not large enough for file name");
```

Comment:
This is the first of many (char*) casts added to this module; their absence causes compiler warnings  i.e. unnecessary and distracting cruft during builds.